geometric_sqrt.go

```diff
@@ -0,0 +1,51 @@
+package interpolator
+
+import (
+	"fmt"
+	"math"
+)
+
+// GeometricSqrt is a geometric interpolator scaled with a square root.
```

Comment:
I think you can improve the explanation here, as it's not super clear. I would write something like:
```
// GeometricSqrt performs a geometric interpolation with respect to the square root of the abscissae
```